### PR TITLE
Move to namespaced phpunit versions

### PR DESCRIPTION
**Pull request by DavidPrevot**
These changes should be needed when you decide to test the code with PHPUnit 6.

Bug-Debian: https://bugs.debian.org/882930

### CODE REVIEW

**Comment by carusogabriel** [comment]
Some tests were declared wrong.

**Comment by keradus** [comment]
what about php 5.3.3 ?

File: src/Util/TestListener.php [hunk 1]
```diff
@@ -11,15 +11,22 @@
...
+use PHPUnit\Framework\SkippedTestError;
+use PHPUnit\Framework\Test;
+use PHPUnit\Framework\TestListener as PHPUnit_Framework_TestListener;
+use PHPUnit\Framework\TestSuite;
+
 /**
  * @author Nicolas Grekas <p@tchwork.com>
  */
-class TestListener extends \PHPUnit_Framework_TestSuite implements \PHPUnit_Framework_TestListener
+class TestListener extends TestSuite implements PHPUnit_Framework_TestListener
```
Comment (stof) [inline src/Util/TestListener.php]: This test listener is now totally incompatible with older versions of PHPUnit. This is not good (Symfony itself relies on support for older versions due to PHP support).

File: tests/Iconv/IconvTest.php [hunk 2]
```diff
@@ -110,7 +111,8 @@ public function testIconvMimeDecode()
     public function testIconvMimeDecodeIllegal()
     {
         iconv_mime_decode('Legal encoded-word: =?utf-8?Q?*?= .');
-        $this->setExpectedException('PHPUnit_Framework_Error_Notice', 'Detected an illegal character in input string');
+        $this->expectException('PHPUnit\Framework\Error\Notice');
```
Comment (stof) [inline tests/Iconv/IconvTest.php]: this is not compatible with PHPUnit 4.8, which is needed to test the package on PHP 5.3 to 5.5 (which must remain supported for us)

**Comment by nicolas-grekas** [comment]
Thanks you @DavidPrevot for raising the topic and proposing a patch. I took over in #112

**Pull request closed without merging**